### ISSUE TITLE
Bug: Rule "no-unused-variables" does not recognize variables used in scenario name, step tables, or step strings

### ISSUE THREAD

**Issue by ajthesnare** [vsiakka/gherkin-lint#124]
Rule "no-unused-variables" does not recognize when variables are used in:
1. scenario names
2. step tables
3. step strings

Examples:
```
Feature: Rule "no-unused-variables" bugs

  Scenario Outline: Examples variable <variable> is in the scenario name
    Given this
    When I do that
    Then something should happen
    Examples:
      | variable |
      | foo      |
      | bar      |

  Scenario Outline: Examples variable is in a step table
    Given this:
      | <variable> |
    When I do that
    Then something should happen
    Examples:
      | variable |
      | foo      |
      | bar      |

  Scenario Outline: Examples variable is in a step string
    Given this
    When I do that
    Then this should display:
      """
        <variable>
      """
    Examples:
      | variable |
      | foo      |
      | bar      |
```

Result:
```
node ./node_modules/gherkin-lint ./features/testRule.feature -c ./.gherkin-lintrc
.../cucumber/features/testRule.feature
  8     Examples table variable "variable" is not used in any step    no-unused-variables
  18    Examples table variable "variable" is not used in any step    no-unused-variables
  30    Examples table variable "variable" is not used in any step    no-unused-variables
```

For `Scenario Outline: Examples variable <variable> is in the scenario name`, I'm not sure if this is best practice.  My team seems to have gotten creative with naming conventions and this actually makes the report more clear with dynamic naming.  I'll leave it up to the community to decide if this should be supported.

The other two examples seems clear cut bugs to me.

**Comment by vsiakka** [comment]
Thanks for the bug report! 
I wasn't aware that you can use step arguments, or variables in scenario names. Since Gherkin/cucumber support both of those operations, the linter should support it too. 
I put up a fix in #128 (I used your examples in the unit tests so tyvm for those!)

**Comment by vsiakka** [comment]
Fix published with v2.13.2
